src/neuro_relay/utils.py

```
@@ -0,0 +1,208 @@
+    # Get the directory of this script
+    current_dir = Path(__file__).resolve().parent
+
+    # Walk upward until we find "neuro-relay" folder (the project root)
+    for parent in current_dir.parents:
+        if parent.name == "neuro_relay":
+            project_root = parent
+            break
+    else:
+        raise RuntimeError(
+            "Could not locate 'windows-api' directory above this file.",
+        )
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)